### DISCUSSION TITLE
Get a list of matches instead of letting metadata plugin importing tags automatically

### DISCUSSION THREAD

**Discussion by Bareus** [Difegue/LANraragi#947] [Ideas]
### The issue

I have a fairly large doujin collection (1000+) and want to use the batch operation to auto-tag it. But none of the metadata plugin seems to be accurate enough (for my collection) to always find the correct archive, which can lead to an archive being tagged with a completely wrong match. Since I also have the option `Allow Plugins to replace archive titles` enabled, an archive can also possible be renamed to something completely different than it's original titel. This leads to the fustrating workflow of me having to go through all doujins anyway, check if they are still correctly titled, and if not correct them manually.

### Suggestion

Since I can't trust metadata plugins to be 100% accurate, I'd rather get a list of suggested matches for a given archive. With "matches" I mean if a metadata plugin searches for an archive, and the search returns multiple results (matches), the plugin should return all matches. If I - as a user - get this list of matches I can choose for myself which of the matches (or none) I want to import to the archive.
This can even be extended by allowing a user to select multiple metadata plugins to be used to get a list of matches from them all.

In the case of the batch operation it could save the matches (if there are multiple) until later on a user comes around and select a fitting match.

**Comment by Difegue** [comment]
Hmm, I agree that this'd be a nice thing to have, but it'd be a non-substantial amount of work to overhaul the plugin system to allow for multiple results, then provide the UI and database functionality to store matches and allow users to pick/discard them...  

This is a niche that could be mostly filled by a third-party client, since you can call the [Plugin API](https://sugoi.gitbook.io/lanraragi/v/dev/api-documentation/miscellaneous-other-api) to run metadata plugins and get the resulting tags without modifying the currently saved tags.
